include/mbedtls/error.h

```diff
@@ -114,25 +114,47 @@ extern "C" {
 #define MBEDTLS_ERR_ERROR_GENERIC_ERROR       -0x0001  /**< Generic error */
 #define MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED -0x006E  /**< This is a bug in the library */
 
+/**
+ * \brief Combines a high-level and low-level error code together.
+ *
+ *        Wrapper function for mbedtls_err_add_ext(). See that function for
+ *        more details.
+ */
+#define MBEDTLS_ERROR_ADD( high, low ) \
+        mbedtls_error_add_ext( high, low, __FILE__, __LINE__ )
+
+/**
+ * \brief Testing hook called before adding/combining two error codes together.
+ *        Only used when invasive testing is enabled via MBEDTLS_TEST_HOOKS.
+ */
+extern void (*mbedtls_test_hook_error_add)( int, int, const char *, int );
```

Comment:
This definition and the corresponding declaration should be wrapped in `#if defined(MBEDTLS_TEST_HOOKS)`.

Reply:
I agree that this definition should be wrapped however the following declaration (I assume you mean `mbedtls_err_add_ext`) should not be because it needs to be visible to `MBEDTLS_ERROR_ADD` even when `MBEDTLS_TEST_HOOKS` is not defined.

This was precisely to move away from having two different definitions of `MBEDTLS_ERROR_ADD` as we had before, see: https://github.com/ARMmbed/mbedtls/pull/4006#discussion_r576352099.

Reply:
Sorry, I meant “this declaration and the corresponding definition” (or I meant to post this comment on the definition of `mbedtls_test_hook_error_add` in `error.c`. Either way d33be30564b37868dbe7c18a93c590d1da1011a0 does what I meant.